bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/util/ResourceFormatter.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2012, 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - extracted from ModuleResolutionReport / ModuleRevision
+ *******************************************************************************/
+package org.eclipse.osgi.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import org.eclipse.osgi.internal.framework.FilterImpl;
```

Comment:
dependency on framework internal make this difficult to place in supplemental bundle.

This seems a better fit in the `org.eclipse.osgi.container` package